cli/src/diff_util.rs

```diff
@@ -81,20 +87,38 @@ pub struct DiffFormatArgs {
     /// Generate diff by external command
     #[arg(long)]
     pub tool: Option<String>,
+    #[arg(long, requires("tool"), default_value("dir"))]
+    pub tool_mode: DiffToolMode,
```

Comment:
Perhaps, `--tool-mode` shouldn't require `--tool`, and it has to be `Option<DiffToolMode>`, because:
* it makes sense to set the default tool in user config, and select mode at runtime.
* if the tool doesn't support dir-diff, the default would fall back to file-by-file.

Reply:
I made `--tool-mode` optional and it will override the configured value for a tool, but it is still required to name the tool, because otherwise the usage gets confusing.

Reply:
Well, we have a similar problem with `--context` option. A better way to handle these is to warn/error if the selected tool doesn't support these command-line options.

Reply:
Removed the argument and made this a config only setting.